In [1]:
import pandas as pd
import yfinance as yf
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

In [8]:
df_stocks = pd.read_csv('constituents_csv.csv')

In [9]:
df_stocks

Symbol                    Name                  Sector
0      MMM              3M Company             Industrials
1      AOS         A.O. Smith Corp             Industrials
2      ABT     Abbott Laboratories             Health Care
3     ABBV             AbbVie Inc.             Health Care
4     ABMD             ABIOMED Inc             Health Care
..     ...                     ...                     ...
500    YUM         Yum! Brands Inc  Consumer Discretionary
501   ZBRA      Zebra Technologies  Information Technology
502    ZBH  Zimmer Biomet Holdings             Health Care
503   ZION           Zions Bancorp              Financials
504    ZTS                  Zoetis             Health Care

[505 rows x 3 columns]

<h1>Data Wrangling</h1>

In [10]:
df_stocks.drop(columns=['Name','Sector'],axis=1,inplace=True)
df_stocks.set_index('Symbol')
df_stocks = df_stocks.transpose()
df_stocks.columns=df_stocks.iloc[0].values
df_stocks.drop(index='Symbol',inplace=True)
df_stocks=df_stocks.rename(columns={'BRK.B':'BRK-B','BF.B':'BFB'})

In [11]:
df_stocks

Empty DataFrame
Columns: [MMM, AOS, ABT, ABBV, ABMD, ACN, ATVI, ADBE, AAP, AMD, AES, AFL, A, APD, AKAM, ALK, ALB, ARE, ALXN, ALGN, ALLE, LNT, ALL, GOOGL, GOOG, MO, AMZN, AMCR, AEE, AAL, AEP, AXP, AIG, AMT, AWK, AMP, ABC, AME, AMGN, APH, ADI, ANSS, ANTM, AON, APA, AIV, AAPL, AMAT, APTV, ADM, ANET, AJG, AIZ, T, ATO, ADSK, ADP, AZO, AVB, AVY, BKR, BLL, BAC, BAX, BDX, BRK-B, BBY, BIO, BIIB, BLK, BA, BKNG, BWA, BXP, BSX, BMY, AVGO, BR, BFB, CHRW, COG, CDNS, CPB, COF, CAH, KMX, CCL, CARR, CAT, CBOE, CBRE, CDW, CE, CNC, CNP, CTL, CERN, CF, SCHW, CHTR, ...]
Index: []

[0 rows x 505 columns]

<h2>Pseudo Code</h2>

Code needs to loop through all of the column headers (stock ticker names)

Then code should feed each name into yf.Tickers. In the same line, continue with the .history attribute with a period of <b>one day </b> this returns a dataframe with various stock info in multiple columns.

Get the Opening stock price column from the dataframe and make sure it comes in a format that can be put into a list of the same length as the number of stock tickers we have (505)

In [12]:
#run this once per day
current_price = []
for tick in df_stocks.columns:
    current_price.append(yf.Ticker(tick).history(period='1d').Open.values[0])

In [14]:
df_stocks.loc[len(df_stocks.index)] = current_price

In [97]:
# testing which code segement runs faster
%%timeit
yf.Ticker('MSFT').history(period='1d').Open.values.tolist()[0]

194 ms ± 28.6 ms per loop (mean ± std. dev. of 7 runs, 10 loops each)


In [95]:
# testing which code segement runs faster
%%timeit
yf.Ticker('MSFT').history(period='1d').Open.values[0]

184 ms ± 13.4 ms per loop (mean ± std. dev. of 7 runs, 10 loops each)


In [104]:
df_stocks.drop(index=[0,1],inplace=True,axis=0)

In [15]:
df_stocks

MMM    AOS     ABT   ABBV   ABMD     ACN   ATVI    ADBE     AAP    AMD  \
0  151.68  48.86  101.21  95.54  300.2  223.89  82.81  441.98  152.66  78.67   

   ...   WYNN    XEL    XRX    XLNX   XYL    YUM    ZBRA     ZBH  ZION     ZTS  
0  ...  75.99  68.52  16.11  106.75  74.1  91.57  275.75  137.43  32.6  148.81  

[1 rows x 505 columns]